### PR TITLE
feat: export package.json

### PR DESCRIPTION
**Pull request by itsjwala**
## Proposed changes

[//]: # (Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue.)

fixes https://github.com/webdriverio/webdriverio/issues/9970

## Types of changes

[//]: # (What types of changes does your code introduce to WebdriverIO?)
[//]: # (_Put an `x` in the boxes that apply_)

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

## Checklist

[//]: # (_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._)

- [x] I have read the [CONTRIBUTING](https://github.com/webdriverio/webdriverio/blob/main/CONTRIBUTING.md) doc
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] I have added proper type definitions for new commands (if appropriate)

### CODE REVIEW

File: packages/eslint-plugin-wdio/package.json [hunk 1]
```diff
@@ -17,7 +17,8 @@
         "require": "./cjs/index.js"
       },
       "./cjs/index.js"
-    ]
+    ],
+    "./package.json": "./package.json"
```
Comment (christian-bromann) [inline packages/eslint-plugin-wdio/package.json:21]: As I mentioned we don't need it here
```suggestion
    ]
```

**Review by christian-bromann — CHANGES_REQUESTED** [review]
To stay consistent we should do it in all packages besides `eslint-plugin-wdio`

File: packages/eslint-plugin-wdio/package.json (hunk 1 above)
Comment (itsjwala) [reply packages/eslint-plugin-wdio/package.json]: ack, updated!

**Comment by itsjwala** [comment]
@christian-bromann 

when I run the failing command locally it passes without any errors, could you help in what is causing the failure https://github.com/webdriverio/webdriverio/actions/runs/4435002600/jobs/7788093185?pr=9973?
```node
➜  webdriverio git:(export-package_json) ✗ node ./scripts/build.js

Compiling ESM packages
npx tsc -b packages/wdio-types/tsconfig.json packages/wdio-protocols/tsconfig.json packages/wdio-logger/tsconfig.json packages/wdio-utils/tsconfig.json packages/wdio-config/tsconfig.json packages/wdio-repl/tsconfig.json packages/devtools/tsconfig.json packages/webdriver/tsconfig.json packages/webdriverio/tsconfig.json packages/wdio-globals/tsconfig.json packages/wdio-runner/tsconfig.json packages/wdio-local-runner/tsconfig.json packages/wdio-mocha-framework/tsconfig.json packages/wdio-browser-runner/tsconfig.json packages/wdio-cli/tsconfig.json packages/wdio-reporter/tsconfig.json packages/eslint-plugin-wdio/tsconfig.json packages/wdio-allure-reporter/tsconfig.json packages/wdio-appium-service/tsconfig.json packages/wdio-browserstack-service/tsconfig.json packages/wdio-concise-reporter/tsconfig.json packages/wdio-crossbrowsertesting-service/tsconfig.json packages/wdio-cucumber-framework/tsconfig.json packages/wdio-devtools-service/tsconfig.json packages/wdio-dot-reporter/tsconfig.json packages/wdio-firefox-profile-service/tsconfig.json packages/wdio-jasmine-framework/tsconfig.json packages/wdio-junit-reporter/tsconfig.json packages/wdio-sauce-service/tsconfig.json packages/wdio-selenium-standalone-service/tsconfig.json packages/wdio-shared-store-service/tsconfig.json packages/wdio-smoke-test-reporter/tsconfig.json packages/wdio-smoke-test-service/tsconfig.json packages/wdio-spec-reporter/tsconfig.json packages/wdio-static-server-service/tsconfig.json packages/wdio-sumologic-reporter/tsconfig.json packages/wdio-testingbot-service/tsconfig.json packages/wdio-webdriver-mock-service/tsconfig.json

Compiling CJS packages
npx tsc -b packages/wdio-smoke-test-cjs-service/tsconfig.json packages/eslint-plugin-wdio/tsconfig.cjs.json packages/wdio-allure-reporter/tsconfig.cjs.json packages/wdio-globals/tsconfig.cjs.json

Removing `export {}` from CJS files compiled using module ESNext
➜  webdriverio git:(export-package_json) ✗ echo $?
0
```

**Comment by christian-bromann** [comment]
Please update your branch.

File: packages/wdio-config/package.json [hunk 2]
```diff
@@ -6,8 +6,17 @@
...
+  "exports": {
+    ".": [
+      {
+        "import": "./build/index.js",
+        "types": "./build/index.d.ts"
+      },
+      "./build/index.js"
+    ],
+    "./package.json": "./package.json"
+  },
```
[the remark this answers is not in the document]
Comment (itsjwala) [reply packages/wdio-config/package.json]: it would work, I'd missed it - made it uniform for all.

**Pull request merged**